File: 0043-multiply-strings/0043-multiply-strings.cpp
class Solution {
public:
    string multiply(string num1, string num2) {
        
        string res(num1.size()+num2.size(),'0');
        for(int i=num1.size()-1;i>=0;i--){
         int n1=num1[i]-'0',carry=0;
            for(int j=num2.size()-1;j>=0;j--){
                
                int r=(num1[i]-'0')*(num2[j]-'0')+(res[i+j+1]-'0');
                // int r=n1*(num2[j]-'0')+(res[i+j+1]-'0');
                // carry=r/10;r=r%10;
                res[i+j+1]=(r%10)+'0';
                res[i+j]+=r/10;
            }
            
        }
        for(int i=0;i<res.size();i++)if(res[i]!='0')return res.substr(i);
        return "0";
    }
};

File: 0043-multiply-strings/0043-multiply-strings.java
class Solution {
    public String multiply(String num1, String num2) {
        
        int m=num1.length(),n=num2.length();
        int res[]=new int[m+n];
        for(int i=m-1;i>=0;i--){
            for(int j=n-1;j>=0;j--){
                res[i+j+1]+=(num1.charAt(i)-'0')*(num2.charAt(j)-'0');
            }
        }
        int carry=0;
        for(int i=m+n-1;i>=0;i--){
            int temp=(res[i]+carry)%10;
            carry=(res[i]+carry)/10;
            res[i]=temp%10;
        }
        StringBuilder sb=new StringBuilder();
        for(int num:res)sb.append(num);
        while(sb.length()!=0 && sb.charAt(0)=='0')sb.deleteCharAt(0);
        return sb.length()==0?"0":sb.toString();
        
    }
}